src/api/controllers/followController.js

```diff
@@ -0,0 +1,57 @@
+import models from '../models/index';
+import status from '../../helpers/constants/status.codes';
+import errorSender from '../../helpers/error.sender';
+/**
+ * @class
+ */
+export default class Follow {
+  /**
+   * This function add the follow count to the database
+   * @param {*} req request sent by user
+   * @param {*} res response after excution of request
+   * @memberof Follow
+   * @return {object} res
+   */
+  static async follow(req, res) {
+    const { username } = req.params;
+    const follower = req.user.username;
+    try {
+      await models.Follow.create({
+        follower,
+        following: username
+      });
+      res.status(status.CREATED).json({
+        status: 201,
+        message: `You are now following ${username}`
+      });
+    } catch (error) {
+      errorSender(status.BAD_REQUEST, res, 'follow', `Already following ${username}`);
```

Comment:
@truestbyheart how do you know that the user is already following? I don't see a check for this.

Reply:
@MemunaHaruna we implemented this together.
the idea here is that we have as primary key the compound of both `fallower:following`.

It means there is a unique constraint on their values compounded together.

Now if the user tries to follow another twice it will violate that uniqueness and it will raise an exception that @truestbyheart is catching.

Reply:
Since its compound key, it will never allow duplication but I have created a script that checks the error code for duplication and self follow error  just to prevent the user from following himself and trying to follow a user he/she is following